lib/stdlib/src/erl_lint.erl

```diff
@@ -2903,10 +2903,9 @@ expr({ssa_check_when,_Anno,_WantedResult,_Args,_Tag,_Exprs}, _Vt, St) ->
 
 %% Check a call to function without a module name. This can be a call
 %% to a BIF or a local function.
-check_call(Anno, record_info, As, Aa, St0) ->
+check_call(Anno, record_info, 2, As, Aa, St0) ->
```

Comment:
This works, but it complicates the caller and introduces the risk that the `As` and `A` arguments are inconsistent. A simpler fix would be to remove the other changes and change this line like so:

```suggestion
check_call(Anno, record_info, [_,_]=As, Aa, St0) ->
```